### PR TITLE
firefox: 70.0.1 -> 71.0

### PR DESCRIPTION
**Pull request by andir**
##### Motivation for this change

This updates Firefox to the upcoming stable version.

This might look like it should go to staging but it actually just has to land on master AFTER #74795 since that already includes the NSS bump. I kept it in this branch to have a consistent history regarding what kind of updates are required to make this work.

Release notes should pop up at https://www.mozilla.org/en-US/firefox/71.0/releasenotes/ some time today/tomorrow.

I'll start working on a backport of all the deps to stable within the next day.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

**Comment by taku0** [comment]
@andir could you also update `firefox-esr`? If so, I will close #74907 as a duplication.

**Comment by andir** [comment]
> @andir could you also update `firefox-esr`? If so, I will close #74907 as a duplication.

Yes, will do. Will push once the build succeeds.

File: pkgs/applications/networking/browsers/firefox/common.nix [hunk 1]
```diff
@@ -95,7 +95,7 @@ let
   browserPatches = [
     ./env_var_for_system_dir.patch
   ]
-  ++ lib.optional (lib.versionAtLeast ffversion "63" && lib.versionOlder ffversion "69")
+  ++ lib.optional (lib.versionAtLeast ffversion "63" && lib.versionOlder ffversion "68.2.0")
```
Comment (taku0) [inline pkgs/applications/networking/browsers/firefox/common.nix]: This should be "68.3.0". This is my mistake.

**Review by taku0 — APPROVED** [review]
Other than `lib.versionOlder ffversion "68.2.0"`, LGTM and succeeded to build and run.

File: pkgs/applications/networking/browsers/firefox/common.nix (hunk 1 above)
Comment (andir) [reply pkgs/applications/networking/browsers/firefox/common.nix]: Fixed.

**Comment by mweinelt** [comment]
Tested with nix-review and run for an hour without issues.

**Comment by edolstra** [comment]
Can you check if it builds with Rust 1.39? Currently we have a line `inherit (rustPackages_1_38_0) cargo rustc;` in `firefoxPackages` which would be nice to get rid of.

**Comment by andir** [comment]
> Can you check if it builds with Rust 1.39? Currently we have a line `inherit (rustPackages_1_38_0) cargo rustc;` in `firefoxPackages` which would be nice to get rid of.

Firefox will probably work (at least 71.x) but the tor browser will not work just yet.

I am planning on reworking the way we are currently passing packages into this as it isn't very flexible enough for the variants of packages we are building with `firefoxPackages`.

**Comment by FRidh** [comment]
staging-next got merged.

**Comment by fpletz** [comment]
The firefox version bump seems to break indexedDB at least for some extensions and thus prevents saving their preferences on my machine. This has also been reported upstream and websites also seem to be affected though I haven't tested this yet. We should revert the firefox bump ASAP, at least on 19.09!

Fedora has the same problems: https://bugzilla.redhat.com/show_bug.cgi?id=1779762

One of the upstream reports is here: https://bugzilla.mozilla.org/show_bug.cgi?id=1601365

And this might be the root cause: https://bugzilla.mozilla.org/show_bug.cgi?id=1601707. We don't have GCC9 and thus might not be affected by the actual segfault described there but Comment 7 refers that more fixes are needed to get extensions to work.

I will continue to investigate but wanted to share my findings as soon as possible.

**Comment by fpletz** [comment]
Since this has been verified by some other people, I've reverted it on 19.09 now because this change was already part of a 19.09 channel release.

**Comment by Ma27** [comment]
I can confirm on this release-19.09 (in my case one symptom was that riot couldn't decrypt messages in encrypted rooms - in case anyone else stumbles upon this) and the revert fixes the problem.

Folks who use channels rather than the git branches to build their systems (latest channel release for 19.09 was 8 hours ago which means that this package is part of it) can work around the issue by either using `firefox-bin` for now or by including the firefox package from the git branch (e.g. `(import builtins.fetchGit ... {}).firefox`) rather than `pkgs.firefox`, it seems as most of the build artifacts in the binary cache for firefox can be reused, at least I didn't have to rebuild :)

**Pull request merged**